src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -248,7 +248,13 @@ private void OnConnectionChanged(object sender, bool connected)
 
                     // s_lastRemoteClientTask info should be saved in the dump
                     // report NFW when connection is closed unless it is proper shutdown
-                    FatalError.ReportWithoutCrash(new Exception("Connection to remote host closed"));
```

Comment:
It seems like your change would be *almost* equivalent to just adding the following to the beginning of `FatalError.ReportWithoutCrash`:

```csharp
if (exception.StackTrace == null)
{
    try
    {
        throw exception;
    }
    catch
    {
        // Empty... we just needed the exception to have a stack trace
    }
}
```

The only difference between what you wrote and this would be the inclusion of `ReportWithoutCrash` at the top of the stack trace. If that would still work, it definitely makes the code easier to read.